app/src/main/kotlin/at/bitfire/davdroid/db/AppDatabase.kt

```diff
@@ -110,9 +113,40 @@ abstract class AppDatabase: RoomDatabase() {
 
     companion object {
 
+        // automatic migrations
+
+        @VisibleForTesting
+        val autoMigrationSpecs: List<(Context) -> AutoMigrationSpec> = listOf(
+            { AutoMigration11_12(it) }
+        )
+
         // manual migrations
 
         val migrations: Array<Migration> = arrayOf(
+            object : Migration(15, 16) {
+                override fun migrate(db: SupportSQLiteDatabase) {
+                    // The timezone column has been removed, now it's timezoneId.
+                    // First, create the new column to store the migrated values.
+                    db.execSQL("ALTER TABLE collection ADD COLUMN timezoneId TEXT DEFAULT NULL")
+                    // Now, fetch all the timezone values.
+                    db.query("SELECT id, timezone FROM collection").use { cursor ->
+                        while (cursor.moveToNext()) {
+                            val id: Long = cursor.getLong(0)
+                            val timezone: String = cursor.getString(1) ?: continue
+                            val vTimeZone = DateUtils.parseVTimeZone(timezone)
+                            val timezoneId = vTimeZone.timeZoneId.value
+                            db.execSQL("UPDATE collection SET timezoneId=? WHERE id=?", arrayOf(timezoneId, id))
+                        }
+                    }
+                    try {
+                        db.execSQL("ALTER TABLE collection DROP COLUMN timezone")
+                    } catch (_: SQLiteException) {
+                        // DROP COLUMN is not available on old versions of SQLite, so if the drop
+                        // fails, just keep the "timezone" column.
```

Comment:
`DROP COLUMN` is available since SQLite 3.35.0, which is available since API 34.

So could we do a version check here instead of the try/catch? We would then have to test (manually) with 33 and 34 to make sure both are working.

Advantage would be that in some far future when API level 34 is the minimum, we can drop the unused code. However an unconditional try/catch would stay forever.